python/python/lance/ray/sink.py

```diff
@@ -15,10 +27,93 @@
 if TYPE_CHECKING:
     import pandas as pd
 
-__all__ = ["LanceDatasink"]
+__all__ = ["LanceDatasink", "LanceFragmentWriter", "LanceCommitter"]
+
+
+def _pd_to_arrow(
+    df: Union[pa.Table, "pd.DataFrame", Dict], schema: Optional[pa.Schema]
+) -> pa.Table:
+    """Convert a pandas DataFrame to pyarrow Table."""
+    from ..dependencies import _PANDAS_AVAILABLE
+    from ..dependencies import pandas as pd
+
+    if isinstance(df, dict):
+        return pa.Table.from_pydict(df, schema=schema)
+    if _PANDAS_AVAILABLE and isinstance(df, pd.DataFrame):
+        return pa.Table.from_pandas(df, schema=schema)
+    return df
+
+
+def _write_fragment(
+    stream: Iterable[pa.Table], uri: str, schema: Optional[pa.Schema] = None
+) -> Tuple[FragmentMetadata, pa.Schema]:
+    from ..dependencies import _PANDAS_AVAILABLE
+    from ..dependencies import pandas as pd
+
+    if schema is None:
+        first = next(stream)
+        if _PANDAS_AVAILABLE and isinstance(first, pd.DataFrame):
+            schema = pa.Schema.from_pandas(first)
+        else:
+            schema = first.schema
+        stream = chain([first], stream)
+
+    def record_batch_converter():
```

Comment:
pedantic nit:
```python
map(
  functools.partial(_pd_to_arrow, schema=schema),
  stream,
)
```

Reply:
Still need to call `to_batches()` over this generator?

Reply:
oh, I see.

```
chain(map(pa.Table.to_batches, map(partial(_pd_to_arrow, schema=schema), stream)))
```
?

jk, nvm as-is is fine